lexxpluss_apps/src/actuator_controller.cpp

```diff
@@ -92,71 +94,224 @@ enum class POS {
+            }
+
+            // catch rise edge
+            if (instance->ch1_prev == 0 && ch1 == 1) {
+                if (ch2 == 0) {
+                    instance->count++;
+                } else {
+                    instance->count--;
+                }
+            }
+
+            instance->ch1_prev = ch1;
+        }
+    }
+    static void static_center_left_callback(struct k_timer *timer_id) {
+        auto* instance = static_cast<encoder*>(k_timer_user_data_get(timer_id));
+        if (instance) {
+            int ch1{0}, ch2{0};
+            gpio_dt_spec dev_ch1 = GPIO_DT_SPEC_GET(DT_NODELABEL(encoder_left_actuator_ch1), gpios);
+            gpio_dt_spec dev_ch2 = GPIO_DT_SPEC_GET(DT_NODELABEL(encoder_left_actuator_ch2), gpios);
+
+            if(gpio_is_ready_dt(&dev_ch1)) {
+                ch1 = gpio_pin_get_dt(&dev_ch1);
+            }
 
-// This class for encoder count, this calls encoder class
+            if(gpio_is_ready_dt(&dev_ch2)) {
+                ch2 = gpio_pin_get_dt(&dev_ch2);
+            }
+
+            // catch rise edge
+            if (instance->ch1_prev == 0 && ch1 == 1) {
+                if (ch2 == 0) {
+                    instance->count++;
+                } else {
+                    instance->count--;
+                }
+            }
+
+            instance->ch1_prev = ch1;
+        }
+    }
+    static void static_center_right_callback(struct k_timer *timer_id) {
+        auto* instance = static_cast<encoder*>(k_timer_user_data_get(timer_id));
+        if (instance) {
+            int ch1{0}, ch2{0};
+            gpio_dt_spec dev_ch1 = GPIO_DT_SPEC_GET(DT_NODELABEL(encoder_right_actuator_ch1), gpios);
+            gpio_dt_spec dev_ch2 = GPIO_DT_SPEC_GET(DT_NODELABEL(encoder_right_actuator_ch2), gpios);
+
+            if(gpio_is_ready_dt(&dev_ch1)) {
+                ch1 = gpio_pin_get_dt(&dev_ch1);
+            }
 
-// May be we can add software-encoder features here?
+            if(gpio_is_ready_dt(&dev_ch2)) {
+                ch2 = gpio_pin_get_dt(&dev_ch2);
+            }
+
+            // catch rise edge
+            if (instance->ch1_prev == 0 && ch1 == 1) {
+                if (ch2 == 0) {
+                    instance->count++;
+                } else {
+                    instance->count--;
+                }
+            }
+
+            instance->ch1_prev = ch1;
+        }
+    }
+    k_timer encoder_count_c, encoder_count_l, encoder_count_r;
+    int ch1_prev{0}, ch2_prev{0};
+    int16_t count{0};
+    bool is_rise_edge{false};
+};
 
 class counter {
 public:
     int init(POS pos) {
         int result{0};
         switch (pos) {
         case POS::CENTER:
-            result = enc.init(TIM4);
-            mm_per_pulse = 50.0f / 1054.0f;
+            // result = enc.init(TIM4); // for HW counter
+            result = enc.init(pos);
+            // mm_per_pulse = 50.0f / 1054.0f; // for HW counter
+            mm_per_pulse = 50.0f / 144.0f;
```

Comment:
Note that the LexxTug is not 50mm.
This would be a separate Issue/Pull Request to be addressed, if it is to be addressed at all.

LexxTugの場合、50mmではないことに注意が必要です。
これは、対応するとしても別のIssue/Pull Requestでの対応となると思います。